htmd/smallmol/chemlab/builder.py

```diff
@@ -0,0 +1,406 @@
+        """
+
+        ## TODO
+        ## Fix the the kekulization problem whem removing hydrogen from NH in aromatic system.
+        sm = self.smallmol
+        hydrogensIdx =  sm.get('idx', 'element H')
+
+        neighborsIdx = np.concatenate(sm.neighbors[hydrogensIdx])
+        neighbors = sm.element[neighborsIdx]
+        hydrogensPolarIdx = hydrogensIdx[np.where( neighbors != 'C' )]
+        hydrogensNonPolarIdx = hydrogensIdx[np.where(neighbors == 'C')]
+
+
+        hydrogensIdx = np.array([],dtype=np.int)
+        if removePolars:
+            hydrogensIdx = np.append( hydrogensIdx, hydrogensPolarIdx)
+        if removeNonPolars:
+            hydrogensIdx = np.append(hydrogensIdx, hydrogensNonPolarIdx)
+
+        self._removeAtoms(hydrogensIdx)
+
+    def _addAtoms(self, atoms):
+        """
+        Adds the atoms one by one from the list of atoms data passed
+
+        Parameters
+        ----------
+        atoms: list
+            List of the atom data to add
+        """
+
+        sm = self.smallmol
+
+        #c = 1
+        for a in atoms:
+            n_atom = sm.numAtoms
+           # print('placing {} to {}'.format(n_atom, a['attachTo']))
+            sm.__dict__['idx'] = np.append(sm.idx, n_atom)
+            sm.__dict__['element'] = np.append(sm.element, a['element'])
+            sm.__dict__['atomname'] = np.append(sm.atomname, '{}{}'.format(a['element'], n_atom))
+            sm.__dict__['charge'] = np.append(sm.charge, a['charge'])
+            sm.__dict__['formalcharge'] = np.append(sm.formalcharge, a['formalcharge'])
+            sm.__dict__['chiral'] = np.append(sm.chiral, a['chiral'])
+            sm.__dict__['hybridization'] = np.append(sm.hybridization, _hybridizations_IdxToType[a['hybridization']])
+            sm.__dict__['coords'] = np.concatenate((sm.coords, a['coords']), axis=0)
+            sm.__dict__['neighbors'] = np.array(sm.neighbors.tolist() + [[a['attachTo']]])
+            sm.__dict__['bondtypes'] = np.array(sm.bondtypes.tolist() + [[a['bondtype']]])
+            if isinstance(sm.neighbors[a['attachTo']], np.ndarray):
+                new_neighbors = sm.neighbors.tolist()
+                new_neighbors[a['attachTo']].append(n_atom)
+                sm.__dict__['neighbors'] = np.array(new_neighbors)
+                new_bonds = sm.bondtypes.tolist()
+                new_bonds[a['attachTo']].append(a['bondtype'])
+                sm.__dict__['bondtypes'] = np.array(new_bonds)
+                
+            else:
+                sm.neighbors[a['attachTo']].append(n_atom)
+                sm.bondtypes[a['attachTo']].append(a['bondtype'])
+
+            #TODO based on chiral
+            sm.__dict__['_chiraltags'] = np.append(sm._chiraltags, 0)
+            # Development note: removed assignement explicitHs
+
+            # coords
+            sm.__dict__['coords'][n_atom]  = self._getAtomCoords(n_atom, a['attachTo'])
+
+            #c += 1
+
+    def _getAtomCoords(self, atomIdx, attachToIdx):
```

Comment:
Maybe add VSEPR to the function name. Otherwise I would expect it to just return atom coordinates

Reply:
Actually the method is returning the coordinates of that atom. Maybe should be called _computeCoord. I wait your opinion on that

Reply:
Hm yes, compute sounds better if you do such have computation. `_computeAtomCoords` maybe?